File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/packet/CBPacketAbilities.java
package pize.tests.voxelgame.main.net.packet;

import pize.net.tcp.packet.IPacket;
import pize.net.tcp.packet.PacketHandler;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;

import java.io.IOException;

public class CBPacketAbilities extends IPacket<PacketHandler>{
    
    public static final int PACKET_ID = 21;
    
    public CBPacketAbilities(){
        super(PACKET_ID);
    }
    
    
    public boolean flyEnabled;
    
    public CBPacketAbilities(boolean flyEnabled){
        this();
        this.flyEnabled = flyEnabled;
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.writeBoolean(flyEnabled);
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        flyEnabled = stream.readBoolean();
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/packet/CBPacketAbilities.kt
package pize.tests.voxelgame.main.net.packet

import pize.net.tcp.packet.IPacket
import pize.net.tcp.packet.PacketHandler
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException

class CBPacketAbilities() : IPacket<PacketHandler?>(PACKET_ID) {
    var flyEnabled = false

    constructor(flyEnabled: Boolean) : this() {
        this.flyEnabled = flyEnabled
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.writeBoolean(flyEnabled)
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        flyEnabled = stream!!.readBoolean()
    }

    companion object {
        const val PACKET_ID = 21
    }
}
